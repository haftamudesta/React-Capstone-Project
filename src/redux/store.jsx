import { configureStore } from '@reduxjs/toolkit';
import memesReducer from './memeSlice';

const store = configureStore({
  reducer: {
    memes: memesReducer,
  },
});
export default store;
