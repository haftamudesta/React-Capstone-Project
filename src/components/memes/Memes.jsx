import { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { fetchingMemesInfo } from '../../redux/memeSlice';
import MemesCard from './MemesCard';
import classes from './Memes.module.css';

const Memes = () => {
  const navigate = useNavigate();
  const { isLoading, memesData } = useSelector((state) => state.memess);
  const [renderedMemes, setRenderedMemes] = useState([]);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchingMemesInfo());
  }, [dispatch]);

  const navigateDetail = (detail) => {
    if (detail.id) {
      navigate(`/memesDetail/${detail.id}`);
    }
  };

  const filteredMemes = memesData
    ? memesData.filter((item) => {
      if (renderedMemes === '') {
        return item;
      }

      if (item.name.toLowerCase().includes(renderedMemes)) {
        return item;
      }
      return null;
    })
    : [];
  let colorPicker = true;
  let checkColor = 1;

  return (
    <div>
      <div className={classes.title__input}>
        <h1 className={classes.memes__title}>Memes Generator</h1>
        <input
          type="text"
          onChange={(e) => setRenderedMemes(e.target.value)}
          placeholder="Search for a memes..."
          value={renderedMemes}
          className={classes.input__container}
        />
      </div>
      <div className={classes.grid}>
        {isLoading ? (
          <h1>Loading...</h1>
        ) : (
          filteredMemes.length > 0
          && filteredMemes.map((meme, index) => {
            checkColor += 1;
            if (checkColor === 3) {
              colorPicker = !colorPicker;
              checkColor = 1;
            }
            return (
              <div
                key={index}
                className={
                  colorPicker
                    ? classes.memes__container1
                    : classes.memes__container2
                }
                aria-hidden="true"
                onClick={() => navigateDetail(meme)}
              >
                <MemesCard
                  image={meme.url}
                  name={meme.name}
                  boxCount={meme.box_count}
                />
              </div>
            );
          })
        )}
      </div>
    </div>
  );
};

export default Memes;
