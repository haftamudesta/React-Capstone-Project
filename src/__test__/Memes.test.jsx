import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Memes from '../components/memes/memes';
import store from '../redux/store';

it('Test MemesContainer renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Memes />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
