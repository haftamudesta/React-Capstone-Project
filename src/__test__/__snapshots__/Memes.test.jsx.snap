// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MemesContainer renders correctly 1`] = `
<div>
  <div>
    <input
      className="input__container"
      onChange={[Function]}
      placeholder="Search for a memes..."
      type="text"
      value={Array []}
    />
  </div>
</div>
`;
